lib/stitch_fix/log_weasel/middleware.rb

```diff
@@ -14,19 +15,31 @@ def initialize(app, options = {})
     # Future: Maybe add X-Amzn-Trace-Id request header for tracing through load balancer
     # (http://docs.aws.amazon.com/elasticloadbalancing/latest/application/load-balancer-request-tracing.html)
     def call(env)
-      x_correlation_id = env.fetch(REQUEST_ID_KEY, nil) || env.fetch(CORRELATION_ID_KEY, nil)
+      x_correlation_id = env.fetch(REQUEST_ID_KEY, nil) || env.fetch(CORRELATION_ID_KEY, nil) || log_weasel_id_from_params(env)
 
       if x_correlation_id
         LogWeasel::Transaction.id = x_correlation_id
       else
         log_weasel_key = env.fetch("HTTP_#{KEY_HEADER}", @key)
         LogWeasel::Transaction.create(log_weasel_key)
-        env[CORRELATION_ID_KEY] = LogWeasel::Transaction.id
-        env[REQUEST_ID_KEY] = LogWeasel::Transaction.id
       end
+
+      env[CORRELATION_ID_KEY] = LogWeasel::Transaction.id
+      env[REQUEST_ID_KEY] = LogWeasel::Transaction.id
+
       @app.call(env)
     ensure
       LogWeasel::Transaction.destroy
     end
+
+    private
+
+    def log_weasel_id_from_params(env)
+      return unless ENV.fetch('LOG_WEASEL_FROM_PARAMS', nil)
```

Comment:
Updated the env var to match the same case as the method `log_weasel` vs. `logweasel`.

I should probably add something about this to the README as well.